presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -781,24 +781,25 @@ public void createTable(ConnectorSession session, ConnectorTableMetadata tableMe
             tableProperties.put(AVRO_SCHEMA_URL_KEY, validateAndNormalizeAvroSchemaUrl(avroSchemaUrl, hdfsContext));
         }
 
-        // Textfile specific properties
-        getTextHeaderSkipCount(tableMetadata.getProperties()).ifPresent(headerSkipCount -> {
+        // Textfile and CSV specific properties
+        ImmutableSet<HiveStorageFormat> expectedFormats = ImmutableSet.of(HiveStorageFormat.TEXTFILE, HiveStorageFormat.CSV);
+        getHeaderSkipCount(tableMetadata.getProperties()).ifPresent(headerSkipCount -> {
             if (headerSkipCount > 0) {
-                checkFormatForProperty(hiveStorageFormat, HiveStorageFormat.TEXTFILE, TEXTFILE_SKIP_HEADER_LINE_COUNT);
-                tableProperties.put(TEXT_SKIP_HEADER_COUNT_KEY, String.valueOf(headerSkipCount));
+                checkFormatForProperty(hiveStorageFormat, expectedFormats, SKIP_HEADER_LINE_COUNT);
```

Comment:
Maybe:
```
checkFormatForProperty(hiveStorageFormat, HiveStorageFormat.TEXTFILE, TEXTFILE_SKIP_HEADER_LINE_COUNT);
checkFormatForProperty(hiveStorageFormat, HiveStorageFormat.CSV, TEXTFILE_SKIP_HEADER_LINE_COUNT);
```?

Reply:
But a csv format would fail in first statement and similarly for text format.